### ISSUE TITLE
ALT+183 gives ╖ (expected À)

### ISSUE THREAD

**Issue by AlexandreHT** [federico-terzi/espanso#1905]
**Describe the bug**
When espanso runs, the special char for ALT+183 is ```╖``` (not expected). When espanso doesn't run, the special char for ALT+183 is ```À``` (as expected)
(cf. https://github.com/espanso/espanso/issues/988#issuecomment-1942048113)

**To Reproduce**
1.type ALT + 183

**Expected behavior**
Special char ```À```

**Screenshots**
NA

**Logs**
```10:53:04 [daemon(19716)] [INFO] reading configs from: "C:\\Users\\[redacted]\\AppData\\Roaming\\espanso"
10:53:04 [daemon(19716)] [INFO] reading packages from: "C:\\Users\\[redacted]\\AppData\\Roaming\\espanso\\match\\packages"
10:53:04 [daemon(19716)] [INFO] using runtime dir: "C:\\Users\\[redacted]\\AppData\\Local\\espanso"
10:53:04 [daemon(19716)] [INFO] system info: Windows v10 (19045) - kernel: 19045
10:53:04 [daemon(19716)] [INFO] watching for changes in path: "C:\\Users\\[redacted]\\AppData\\Roaming\\espanso"
10:53:04 [daemon(19716)] [INFO] espanso version: 2.2.1
10:53:04 [daemon(19716)] [INFO] spawning the worker process...
10:53:04 [daemon(19716)] [INFO] binded to named pipe: \\.\pipe\espansodaemonv2
10:53:04 [worker(4592)] [INFO] reading configs from: "C:\\Users\\[redacted]\\AppData\\Roaming\\espanso"
10:53:04 [worker(4592)] [INFO] reading packages from: "C:\\Users\\[redacted]\\AppData\\Roaming\\espanso\\match\\packages"
10:53:04 [worker(4592)] [INFO] using runtime dir: "C:\\Users\\[redacted]\\AppData\\Local\\espanso"
10:53:04 [worker(4592)] [INFO] system info: Windows v10 (19045) - kernel: 19045
10:53:04 [worker(4592)] [INFO] binded to named pipe: \\.\pipe\espansoworkerv2
10:53:04 [worker(4592)] [INFO] monitoring the status of the daemon process
10:53:04 [worker(4592)] [INFO] using Win32AppInfoProvider
10:53:04 [worker(4592)] [INFO] using Win32Source
10:53:04 [worker(4592)] [INFO] using Win32Injector
10:53:04 [worker(4592)] [INFO] using Win32Clipboard 
```

**Setup information**
- OS: Windows 10 - 22H2 (build 19045.4046)
- Version: espanso 2.2.1

**Comment by smeech** [comment]
Are you using the `emulate_alt_codes` [option](https://espanso.org/docs/configuration/options/)?

**Comment by smeech** [comment]
According to https://www.alt-codes.net/ and other sources, ALT-183 is "╖". "À" is ALT-0192. Does that work for you?
I see you've been reporting this [problem](https://github.com/espanso/espanso/issues/988#issuecomment-1206777014) for a while!

**Comment by AlexandreHT** [comment]
Yes, ```emulate_alt_codes``` is activated (ALT-144 give ```É``` as expected)
ALT-0192 gives ```À``` but it's only a workaround, since the standard Windows shortcut for À is ALT+183 on my French edition of W10. As I was saying in https://github.com/espanso/espanso/issues/988#issuecomment-1206777014, it looks like espanso implements code page CP437 whereas French windows uses CP1252. Would it be possible to handle this as a parameter?

**Comment by morgan-dgk** [comment]
@AlexandreHT, I am not at all familar with Windows input event handling so someone else may be able to chime in with more experience. That said, it looks like Espanso uses [GetKeyboardLayout](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getkeyboardlayout) on windows to get the input locale identifier. From the linked page, it appears that this is different to the keyboard layout id. I wonder if this is the source of the bug?

**Comment by smeech** [comment]
It appears to be hard-coded [here](https://github.com/espanso/espanso/blob/4e6839c4fa4bb1325d38a5b434dcbc4a61289533/espanso-engine/src/process/middleware/alt_code_synthesizer.rs#L115), but it looks as though there is provision for CP1252.
One for the devs.

**Comment by morgan-dgk** [comment]
Nice find smeech.

Based on this Windows [alt-codes table](https://en.wikipedia.org/wiki/Alt_code), it would seem your version of windows is using CP850. It looks like the alt-code synthesizer linked above should handle both CP1252 and CP437.

**Comment by Braulios1A** [comment]
According to smeech, solution is in "alt_code_synthesizer.rs", could be great define an option menu to select the right code page for each country or region. In my case ALT+155 should be ø and ALT+157 Ø for example. With espanso available return ¢ and ¥.

For any reason I don't know, only in microsoft word works OK with my region alt codes in all other softwares work with espanso alt codes include microsoft excel, powerpoint, outlook.

**Comment by sandubinha** [comment]
Same problem. My code page is also CP850, not supported in [alt_code_synthesizer.rs](https://github.com/espanso/espanso/blob/4e6839c4fa4bb1325d38a5b434dcbc4a61289533/espanso-engine/src/process/middleware/alt_code_synthesizer.rs#L115).
As I use these codes a lot, it's kind of annoying on a daily basis. It's sad to move back to beeftext, at least for a while.

Version: espanso **2.2.3**
